src/ESPiLight.cpp

```diff
@@ -429,3 +449,67 @@ int ESPiLight::stringToPulseTrain(const String &data, uint16_t *codes,
   }
   return -1;
 }
+
+static protocols_t *find_proto(const char *name) {
+  struct protocols_t *pnode = protocols;
+  while (pnode != nullptr) {
+    if (strcmp(name, pnode->listener->id) == 0) {
+      return pnode;
+    }
+    pnode = pnode->next;
+  }
+  return nullptr;
+}
+
+void ESPiLight::limitProtocols(const String &protos) {
+  if (!json_validate(protos.c_str())) {
+    Serial.println("Protocol limit argument is not a valid json message!");
+    return;
+  }
+  JsonNode *message = json_decode(protos.c_str());
+
+  if (message->tag != JSON_ARRAY) {
+    Serial.println("Protocol limit argument is not a json array!");
+    json_delete(message);
+    return;
+  }
+
+  struct protocols_t *pnode = used_protocols;
+  while (pnode != nullptr) {
+    struct protocols_t *tmp = pnode;
+    pnode = pnode->next;
+    free(tmp);
+  }
+
+  used_protocols = nullptr;
+  auto curr = message->children.head;
+
+  while (curr != nullptr) {
+    if (!curr->tag == JSON_STRING) {
+      Serial.println("Element is not a String");
+      continue;
+    }
+
+    auto *templ = find_proto(curr->string_);
+    if (templ == nullptr) {
+      Serial.print("Protocol not found: ");
+      Serial.println(curr->string_);
+      continue;
+    }
+
+    auto *new_node =
+        static_cast<protocols_t *>(malloc(sizeof(struct protocols_t)));
+    new_node->listener = templ->listener;
+    new_node->next = used_protocols;
+    used_protocols = new_node;
+
+    Serial.print("activated protocol ");
+    Serial.println(templ->listener->id);
+
+    if (curr == message->children.tail) {
```

Comment:
This and the condition in the while loop are checking the same?

Reply:
I could not find if `next` is definitely `nullptr` for the last element. So I wanted to make sure that I'm not dereferencing something thats not there.